Client/Frontend/Home/BookmarksEditPanel.swift

```diff
@@ -0,0 +1,335 @@
+    }
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    func tableView(tableView: UITableView, willDisplayHeaderView view: UIView, forSection section: Int) {
+        if let header = view as? BookmarkFolderEditTableViewHeader {
+            // for some reason specifying the font in header view init is being ignored, so setting it here
+            header.textLabel?.font = DynamicFontHelper.defaultHelper.DeviceFontHistoryPanel
+        }
+    }
+    
+    func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
+        return .None
+    }
+    
+    override func tableView(tableView: UITableView, hasFullWidthSeparatorForRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        // Show a full-width border for cells above separators, so they don't have a weird step.
+        // Separators themselves already have a full-width border, but let's force the issue
+        // just in case.
+        let this = self.source?.current[indexPath.row]
+        if (indexPath.row + 1) < self.source?.current.count {
+            let below = self.source?.current[indexPath.row + 1]
+            if this is BookmarkSeparator || below is BookmarkSeparator {
+                return true
+            }
+        }
+        return super.tableView(tableView, hasFullWidthSeparatorForRowAtIndexPath: indexPath)
+    }
+    
+    func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
+        // Intentionally blank. Required to use UITableViewRowActions
+    }
+}
+
+class BookmarksEditPanelCell: UITableViewCell {
+    lazy var label = UILabel()
+    lazy var textField = UITextField()
+    
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        addSubview(label)
+        addSubview(textField)
+        
+        label.font = DynamicFontHelper.defaultHelper.DeviceFontSmallHistoryPanel
+        textField.font = DynamicFontHelper.defaultHelper.DeviceFontHistoryPanel
+        
+        label.snp_makeConstraints { make in
+            make.left.equalTo(self.snp_left).offset(BookmarksEditPanelUX.LabelLeftRightPadding)
+            make.right.equalTo(self.snp_right).offset(-BookmarksEditPanelUX.LabelLeftRightPadding)
+            make.top.equalTo(self.snp_top).offset(BookmarksEditPanelUX.LabelTopBottomPadding)
+        }
+        textField.snp_makeConstraints { make in
+            make.top.equalTo(label.snp_bottom).offset(BookmarksEditPanelUX.TextFieldTopBottomPadding)
+            make.left.equalTo(label.snp_left)
+            make.right.equalTo(label.snp_right)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class ButtonPanelCell: UITableViewCell {
```

Comment:
I don't think you need this. For example, the `DisconnectSetting` for displaying the Log Out button setting looks just like this but uses the default `UITableViewCell` with a colored, centered title. Instead of the button, we can handle the taps using `didSelectRow...`